### PR TITLE
fix #146 empty the graph after delete

### LINKED ISSUE #146: after delete delete graph
after delete the graph should be empty

### PR DESCRIPTION
**Pull request by Anchel123**
## Summary by CodeRabbit

- **Bug Fixes**
	- Corrected import issues and enhanced the graph deletion feature for improved usability.

### CODE REVIEW

File: app/graph/page.tsx [hunk 1]
```diff
@@ -83,7 +83,12 @@ export default function Page() {
 
     return (
         <div className="h-full flex flex-col p-2 gap-y-2">
-            <Query className="border rounded-lg border-gray-300 p-2" onSubmit={runQuery} onQueryUpdate={(state) => { queryState.current = state }} />
+            <Query
+                className="border rounded-lg border-gray-300 p-2"
+                onSubmit={runQuery}
+                onQueryUpdate={(state) => { queryState.current = state }}
+                deleteGraph={() => setGraph(Graph.empty())}
```
Comment (gkorland) [inline app/graph/page.tsx:90]: Please rename to onDeleteGraph.
```suggestion
                onDeleteGraph={() => setGraph(Graph.empty())}
```

Why do you need to empty the Graph and not just call  `setGraph(null)`

File: app/graph/query.tsx [hunk 2]
```diff
@@ -18,9 +18,10 @@ export class QueryState {
     ) { }
 }
 
-export function Query({ onSubmit, onQueryUpdate, className = "" }: {
+export function Query({ onSubmit, onQueryUpdate, deleteGraph, className = "" }: {
     onSubmit: (event: React.FormEvent<HTMLFormElement>) => void,
     onQueryUpdate: (state: QueryState) => void,
+    deleteGraph: () => void,
```
Comment (gkorland) [inline app/graph/query.tsx:24]:
```suggestion
    onDeleteGraph: () => void,
```

File: app/graph/query.tsx [hunk 3]
```diff
@@ -40,6 +41,8 @@ export function Query({ onSubmit, onQueryUpdate, className = "" }: {
             })
             setOnDelete(prev => !prev)
             setGraphName('')
+            setQuery('')
+            deleteGraph()
```
Comment (gkorland) [inline app/graph/query.tsx:45]:
```suggestion
            onDeleteGraph()
```

File: app/graph/query.tsx [hunk 4]
```diff
@@ -57,8 +60,14 @@ export function Query({ onSubmit, onQueryUpdate, className = "" }: {
...
             <div className="flex flex-row space-x-3 w-full md:w-8/12 items-center">
-                <Input id="query" className="border-gray-500 w-full"
-                    placeholder="MATCH (n)-[e]-() RETURN n,e limit 100" type="text" onChange={(event) => setQuery(event.target.value)} />
+                <Input
+                    id="query"
+                    className="border-gray-500 w-full"
+                    placeholder="MATCH (n)-[e]-() RETURN n,e limit 100"
+                    type="text"
+                    onChange={(event) => setQuery(event.target.value)}
+                    value={query}
```
Comment (gkorland) [inline app/graph/query.tsx:69]:
```suggestion

```
Why did you add the `value={query}`?

**Review by gkorland — CHANGES_REQUESTED** [review]

**Review by AviAvni — DISMISSED** [review]

**Review by AviAvni — APPROVED** [review]

**Pull request merged**